Add DMN 1.5 boxed expressions documentation (#108)
IBM/bamoe-docs: runtime-services-modeling/decisions-with-dmn.adoc

For the `some` loop, the right part of the `some` displays the iterator variable name.
The second row holds an expression representing the collection that will be iterated over.
The expression in the `in` row must resolve to a collection.
The last line is an expression that will be evaluated on each item.
The expression defined in the `satisfies` must resolve to a boolean.

.Some expression of Will Not Fly Together decision
image::runtime-services-modeling/dmn-some-expression.png[]

NOTE: This feature was introduced as part of the {PRODUCT} 9.1 and DMN 1.5 releases.

[id="con-dmn-filter-expressions"]
==== Boxed filter expressions

[role="_abstract"]
A boxed filter offers a visual representation of collection filtering.
The top part is an expression that is the collection to be filtered.
The bottom part, between the square brackets, holds the filter expression.

The example below is an alternative way of defining logic as <<flight-capacity-example>>.

.Filter expression of rebooked flights
image::runtime-services-modeling/dmn-filter-expression.png[]

Furthermore, the `filter` expression brings a reserved keyword `item`.

> Using this keyword you can explicitly access items of the filtered collection.

Reviewer: ```suggestion
Using this keyword you can explicitly access items from the filtered collection.
```

[id="approved-credit-score-agency-list-example"]
.Boxed list expression using FEEL logic for approved credit score agency status
image::dmn-list-expression-example2.png[]

.DMN diagram for approved credit score agency status
image::dmn-list-expression-example2a.png[]

[id="con-dmn-conditional-expressions"]
==== Boxed conditional expressions

[role="_abstract"]
A boxed conditional offers a visual representation of an `if` statement using three rows.
The expression in the `if` part must resolve to a boolean.
When the `if` resolves to a `true`, `then` part is returned.
When the `if` resolves to a `false`, `else` part is returned.

.Conditional expression of MonthlyFee computation
image::runtime-services-modeling/dmn-conditional-expression.png[]

NOTE: This feature was introduced as part of {PRODUCT} 9.1 and DMN 1.5 releases.

[id="con-dmn-for-expressions"]
==== Boxed iterator expressions

[role="_abstract"]
A boxed iterator offers a visual representation of an iterator statement.
For the `for` loop, the right part of the `for` displays the iterator variable name.

> The second row holds an expression representing the collection that will be iterated over.

Reviewer: ```suggestion
The second row must contain an expression that resolves to a collection, which will be iterated over.
```

Reviewer: Or 
The `in` row must contain an expression that resolves to a collection, which will be iterated over. 
Just personal preference here, but at first glance it sounded like the second row and the in row were different rows.

> The expression in the `in` row must resolve to a collection.

Reviewer: ```suggestion
```

The last row contains the expression that will process each element of the collection.

The example below is an alternative way of producing list similar as <<approved-credit-score-agency-list-example>>.

.Iterator expression of Approved credit score agencies
image::runtime-services-modeling/dmn-iterator-expression.png[]

NOTE: This feature was introduced as part of the {PRODUCT} 9.1 and DMN 1.5 releases.

[id="con-dmn-every-expressions"]
==== Boxed every expressions

[role="_abstract"]

> A boxed iterator offers a visual representation of an iterator statement.

Reviewer 2: It has the same text of the `Boxed iterator expressions` section, maybe changing the previous section to `Boxed for expression`? And be more specific on the paragraph as this is related to the `every` expression.

`A boxed every offers a visual representation of an iterator statement`

Author: done

For the `every` loop, the right part of the `every` displays the iterator variable name.
The second row holds an expression representing the collection that will be iterated over.
The expression in the `in` row MUST resolve to a collection.
The last line is an expression that will be evaluated on each item.
The expression defined in the `satisfies` must resolve to a boolean.

> .Every expression of Will Flight Together decision

Reviewer: Probably not a big deal, but should be "Will Fly Together" / "Will Not Fly Together".

image::runtime-services-modeling/dmn-every-expression.png[]

NOTE: This feature was introduced as part of the {PRODUCT} 9.1 and DMN 1.5 releases.

[id="con-dmn-some-expressions"]
==== Boxed some expressions

[role="_abstract"]

> A boxed iterator offers a visual representation of an iterator statement.

Reviewer 2: Same here.

Author: done

For the `some` loop, the right part of the `some` displays the iterator variable name.
The second row holds an expression representing the collection that will be iterated over.
The expression in the `in` row must resolve to a collection.
The last line is an expression that will be evaluated on each item.
The expression defined in the `satisfies` must resolve to a boolean.

.Some expression of Will Not Flight Together decision
image::runtime-services-modeling/dmn-some-expression.png[]

NOTE: This feature was introduced as part of {PRODUCT} 9.1 and DMN 1.5 releases.

[id="con-dmn-filter-expressions"]
==== Boxed filter expressions

[role="_abstract"]
A boxed filter offers a visual representation of collection filtering.

> The top part is an expression that is the collection to be filtered.

Reviewer: ```suggestion
The top part contains an expression representing the collection to be filtered.
```

The bottom part, between the square brackets, holds the filter expression.

> The example below is an alternative way of defining logic as <<flight-capacity-example>>.

Reviewer: ```suggestion
The example below is an alternative way of defining logic similar to the <<flight-capacity-example>> example.
```
